src/index.ts

```diff
@@ -61,44 +60,50 @@ import {
   convertApiCountry,
   convertSdkPriceList,
 } from './helpers/converters'
+import { MetadataCodec } from './metadata'
 
 export * from './exceptions'
 export * from './interfaces'
-
-export interface PaddleSdkOptions {
-  publicKey: string
-  vendorId: number
-  vendorAuthCode: string
-  metadataEncryptionKey: string
+export * from './metadata'
+
+export interface PaddleSdkOptions<TMetadata = unknown> {
+  /** Public key from the paddle dashboard to validate webhook requests */
+  readonly publicKey: string
+  /** Vendor ID from the paddle dashboard to authenticate API requests */
+  readonly vendorId: number
+  /** Vendor auth code from the paddle dashboard to authenticate API requests */
+  readonly vendorAuthCode: string
+  /**
+   * Metadata codec encodes and decodes additional pass-through data for an order
+   *
+   * @see https://developer.paddle.com/guides/how-tos/checkout/pass-parameters
+   */
+  readonly metadataCodec: MetadataCodec<TMetadata>
 }
 
-export class PaddleSdk<TMetadata = any> {
+export class PaddleSdk<TMetadata = unknown> {
   private readonly publicKey: string
   private readonly vendorId: number
   private readonly vendorAuthCode: string
-  private readonly metadataEncryptionKey: string
+  private readonly metadataCodec: MetadataCodec<TMetadata>
 
-  constructor(options: PaddleSdkOptions) {
-    if (!options.publicKey) {
+  constructor({ publicKey, vendorId, vendorAuthCode, metadataCodec }: PaddleSdkOptions<TMetadata>) {
+    if (!publicKey) {
       throw new PaddleSdkException('PaddleSdk was called without a publicKey')
     }
-
-    if (!options.vendorId) {
+    if (!vendorId) {
       throw new PaddleSdkException('PaddleSdk was called without a vendorId')
     }
-
-    if (!options.vendorAuthCode) {
+    if (!vendorAuthCode) {
       throw new PaddleSdkException('PaddleSdk was called without a vendorAuthCode')
     }
-
-    if (!options.metadataEncryptionKey || options.metadataEncryptionKey.length !== 32) {
-      throw new PaddleSdkException('PaddleSdk was called with an invalid metadataEncryptionKey')
+    if (!metadataCodec) {
+      throw new PaddleSdkException('PaddleSdk was called with an invalid metadata codec')
```

Comment:
For consistency with the other messages

```suggestion
      throw new PaddleSdkException('PaddleSdk was called without a metadataCodec')
```